### PR TITLE
Dx 252 battery level trait

### PR DESCRIPTION
**Pull request by rlperez**
Dependent on merge of SDK merge and pubspec version for it being updated.

### CODE REVIEW

File: lib/providers/power_trait_provider.dart [hunk 1]
```diff
@@ -7,13 +8,11 @@ typedef GetDeviceDetailsMethod = Future<Device> Function(
 typedef SendPowerMethod = Future<void> Function(
     Request request, String id, bool onOff);
 
-enum PowerState { idle, loading, performingAction, error }
```
Comment (ses110) [inline lib/providers/power_trait_provider.dart:10]: +1 on this. We can probably generalize this State enum to all "Providers" (ChangeNotifiers)

Comment (rlperez) [reply lib/providers/power_trait_provider.dart:10]: Agreed. We also can probably factor out the state as a whole in an abstract provider but I was getting weary of touching too much right now.

File: lib/providers/battery_level_provider.dart [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      {GetDeviceDetailsMethod getDetails =
+          DevicesRepository.getDeviceDetails}) {
+    this._deviceId = deviceId;
+    this._request = request;
+    fetchData(getDeviceDetails: getDetails);
+  }
+
+  Device? get deviceDetail => _deviceDetail;
+
+  Future<Device?> fetchData(
```
Comment (surajadh) [inline lib/providers/battery_level_provider.dart:27]: This probably should be void. I notice same issue with power trait as well provider as well. This method should just set deviceDetail and notify

File: lib/providers/battery_level_provider.dart [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      {GetDeviceDetailsMethod getDeviceDetails =
+          DevicesRepository.getDeviceDetails}) async {
+    _setState = WidgetState.loading;
+
+    try {
+      _deviceDetail = await getDeviceDetails(_request, _deviceId);
+      _setState = WidgetState.idle;
+    } catch (error) {
+      _setErrorState(error.toString());
+      return null;
```
Comment (surajadh) [inline lib/providers/battery_level_provider.dart:37]: This return seems redundant as well

**Review by surajadh — APPROVED** [review]

**Pull request merged**